contracts/peripheral/YieldStreamer.sol

```diff
@@ -0,0 +1,447 @@
+                if (recipientIdsArray[i] == id_) {
+                    recipientIdsArray[i] = recipientIdsArray[recipientIdsArray.length - 1];
+                    recipientIdsArray.pop();
+                    break;
+                }
+            }
+
+            IERC20(gOHM).safeTransfer(msg.sender, principal);
+            IERC20(gOHM).safeTransfer(recipient, totalGOHM - principal);
+            if (unclaimedDai != 0) {
+                IERC20(DAI).safeTransfer(recipient, unclaimedDai);
+            }
+        } else {
+            _withdrawPrincipal(id_, amount_);
+            IERC20(gOHM).safeTransfer(msg.sender, amount_);
+        }
+
+        emit Withdrawn(msg.sender, amount_);
+    }
+
+    /**
+        @notice Withdraw excess yield from your deposit in gOHM.
+        @dev  Use withdrawYieldAsDai() to withdraw yield as DAI.
+        @param id_ Id of the deposit.
+    */
+    function withdrawYield(uint256 id_) external override {
+        if (withdrawDisabled) revert WithdrawDisabled();
+        if (depositInfo[id_].depositor != msg.sender && recipientInfo[id_].recipientAddress != msg.sender)
+            revert UnauthorisedAction();
+
+        recipientInfo[id_].lastUpkeepTimestamp = block.timestamp;
+
+        uint256 yield = _redeemYield(id_);
+
+        IERC20(gOHM).safeTransfer(msg.sender, yield);
+    }
+
+    /**
+        @notice Withdraw excess yield from your deposit in DAI
+        @param id_ Id of the deposit
+    */
+    function withdrawYieldAsDai(uint256 id_) external override {
+        if (withdrawDisabled) revert WithdrawDisabled();
+        if (depositInfo[id_].depositor != msg.sender && recipientInfo[id_].recipientAddress != msg.sender)
+            revert UnauthorisedAction();
+
+        recipientInfo[id_].lastUpkeepTimestamp = block.timestamp;
+
+        uint256 gOHMYield = _redeemYield(id_);
+        uint256 totalOhmToSwap = staking.unwrap(address(this), gOHMYield);
+        staking.unstake(address(this), totalOhmToSwap, false, false);
+
+        IERC20(OHM).approve(address(sushiRouter), totalOhmToSwap);
+        uint256[] memory calculatedAmounts = sushiRouter.getAmountsOut(totalOhmToSwap, sushiRouterPath);
+        uint256[] memory amounts = sushiRouter.swapExactTokensForTokens(
+            totalOhmToSwap,
+            (calculatedAmounts[1] * (1000 - maxSwapSlippagePercent)) / 1000,
+            sushiRouterPath,
+            msg.sender,
+            block.timestamp
+        );
+
+        uint256 daiToSend = recipientInfo[id_].unclaimedDai + amounts[1];
+        recipientInfo[id_].unclaimedDai = 0;
+        IERC20(DAI).safeTransfer(msg.sender, daiToSend);
+    }
+
+    /**
+        @notice harvest all your unclaimed Dai
+        @param id_ Id of the deposit
+    */
+    function harvestDai(uint256 id_) external override {
+        if (withdrawDisabled) revert WithdrawDisabled();
+        if (depositInfo[id_].depositor != msg.sender && recipientInfo[id_].recipientAddress != msg.sender)
```

Comment:
I thought the point of `recipientAddress` is so yield can be sent elsewhere?  Am I misreading this or does this enforce `recipientAddress` is the same as depositor?

Reply:
Yea i shoulda done just recipient.